### DISCUSSION TITLE
Hydromt is unable to infer the flow direction file

### DISCUSSION THREAD

**Discussion by AldenShadd1000** [Deltares/hydromt_wflow#301] [Q&A]
I generated a flow direction file from a DEM of resolution of 90 metres using the code provided at Hydromt website: (https://deltares.github.io/hydromt_wflow/latest/_examples/prepare_ldd.html) But, I encountered the following error:
![Error_Screenshot](https://github.com/user-attachments/assets/14fac9e1-8c94-4ced-85e2-b1ee49e5e23e)

I am attaching the log file below.
[hydromt.log](https://github.com/user-attachments/files/17365804/hydromt.log)

Kindly guide on how to proceed on this.

**Comment by AldenShadd1000** [comment]
Hi, it’s been 5 days since I posted my query, and I haven't received a response yet. Could someone please assist me with the error I encountered while generating the flow direction file using the code from the Hydromt website? I’ve attached the error screenshot and log file for reference.

Thank you!

**Comment by shartgring** [comment]
Dear @AldenShadd1000, thank you for your question. Could you provide the full line of code that you are using to build the model? I suspect that the DataCatalog is missing. See also the final step in https://deltares.github.io/hydromt_wflow/latest/_examples/prepare_ldd.html#Exporting-the-newly-created-data-and-corresponding-data-catalog : here you create a DataCatalog, and in the model builder you can link the path to it using `-d path_to_catalog.yml`. I see in the log that you provided that it is know only reading the DataCatalog from the artifact_data which is used to build the example model

**Reply by AldenShadd1000** [reply]
Thanks for the response.

The complete hydromt command that I used is :
`!hydromt build wflow "./Talwarapar_90m_res" -r "{'subbasin': [74.402086,31.682276], 'strord': 11, 'bounds': [74.37, 31.65, 74.445, 31.72]}" -i wflow_build.ini -d G:/Talwarapar_Water_Residual_Content/Model_Talwarapar/artifact_data/data_catalog.yml -vv`

I have prepared files for my model area and gave those paths accordingly. I have numerous models but in this case I lowered the resolution to 250 meters, and I have encountered this issue.

**Reply by JoostBuitink** [reply]
Not sure if this is causing the issue, but it looks like the bounds you provide create a very small bbox. This bbox should cover the full region you expect to simulate, which (I assume based on the outlet) should be a lot larger than what you are currently providing. 

In addition, the strord condition is rather high. This might not be a problem, but using a lower value might help as well. It only filters the outlet location to a river with at least that streamorder, so a lower value should also work. It is mostly there to prevent accidental snapping to very small rivers.

So perhaps adjusting these two might help as well!

**Reply by AldenShadd1000** [reply]
Thanks for your response.

I have tried adjusting these two but the issue still persists.

**Comment by shartgring** [comment]
Would it be possible for you to use the following notebook (https://github.com/Deltares/pyflwdir/blob/main/examples/basins.ipynb) to see if it is possible to delineate a basin with the given flowdirection data and the bbox? I would suggest loading the data using rasterio, clipping it given the bbox and then following the notebook with the outlet as provided by your example. As under the hood, HydroMT uses PyFlwDir to delineate the basin.

**Reply by AldenShadd1000** [reply]
Thank you for you response.

**Comment by teiushi** [comment]
@AldenShadd1000，I've gotten this error before, and adjusting the flwdir data to unit8 solved it .

**Reply by AldenShadd1000** [reply]
Sorry for late reply!

Can you please explain it in more detail, it will be very helpful.

Thanks

**Reply by teiushi** [reply]
When I use the flwdir(before).tif file:  
<img width="204" alt="be" src="https://github.com/user-attachments/assets/e4beba4c-1bcb-4e01-ad6a-fbf5556ef627" />  
At that time the error was reported.

After that the flwdir(after).tif was regenerated:    
<img width="195" alt="af" src="https://github.com/user-attachments/assets/1da9ff17-895f-49a6-9294-0d8aaac48328" />
This did not give an error and ran successfully.
